src/normalize/normalize.py

```diff
@@ -0,0 +1,109 @@
+import click
+import boto3
+import os
+from datetime import datetime
+from gtfs_realtime_pb2 import FeedMessage
+from src.normalize.protobuf_utils import protobuf_objects_to_pyarrow_table
+from src.normalize.parquet_utils import write_data, add_time_columns
+import time
+import json
+
+
+def get_last_processed_timestamp(s3, bucket, key):
+    try:
+        response = s3.get_object(Bucket=bucket, Key=key)
+        return json.loads(response['Body'].read())['last_processed']
+    except s3.exceptions.NoSuchKey:
+        return None
+
+
+def update_last_processed_timestamp(s3, bucket, key, timestamp):
+    s3.put_object(
+        Bucket=bucket,
+        Key=key,
+        Body=json.dumps({'last_processed': timestamp})
+    )
+
+
+def normalize_raw_feed(raw_input, output_prefix, cur_time, cur_date):
+    feed = FeedMessage()
+    feed.ParseFromString(raw_input)
+
+    trip_updates = []
+    vehicles = []
+    alerts = []
+
+    for entity in feed.entity:
+        entity_id = entity.id
+        if entity.HasField('trip_update'):
+            trip_updates.append((entity_id, entity.trip_update))
+        if entity.HasField('alert'):
+            alerts.append((entity_id, entity.alert))
+        if entity.HasField('vehicle'):
+            vehicles.append((entity_id, entity.vehicle))
+
+    trip_updates_pa = protobuf_objects_to_pyarrow_table([x[1] for x in trip_updates]) if trip_updates else None
+    vehicles_pa = protobuf_objects_to_pyarrow_table([x[1] for x in vehicles]) if vehicles else None
+    alerts_pa = protobuf_objects_to_pyarrow_table([x[1] for x in alerts]) if alerts else None
+
+    if trip_updates_pa:
+        trip_updates_pa = trip_updates_pa.add_column(0, "id", [[x[0] for x in trip_updates]])
+        trip_updates_pa = add_time_columns(trip_updates_pa, cur_time, cur_date)
+        s3_uri = f"{output_prefix}/trip-updates"
+        write_data(trip_updates_pa, s3_uri)
+
+    if vehicles_pa:
+        vehicles_pa = vehicles_pa.add_column(0, "id", [[x[0] for x in vehicles]])
+        vehicles_pa = add_time_columns(vehicles_pa, cur_time, cur_date)
+        s3_uri = f"{output_prefix}/vehicles"
+        write_data(vehicles_pa, s3_uri)
+
+    if alerts_pa:
+        alerts_pa = alerts_pa.add_column(0, "id", [[x[0] for x in alerts]])
+        alerts_pa = add_time_columns(alerts_pa, cur_time, cur_date)
+        s3_uri = f"{output_prefix}/alerts"
+        write_data(alerts_pa, s3_uri)
+
+
+def parse_files(source_prefix, destination_prefix, state_file):
+    s3 = boto3.client('s3')
+
+    source_bucket, source_key = source_prefix.replace("s3://", "").split("/", 1)
```

Comment:
Part of the prefix will be the date, ie s3://gtfs-xxxx/FEED_ID/20240920
Both gtfs-xxx and feed_id should be given but date needs to be inferred so that we dont need to restart the normalizer daily at 00:00 UTC.